docker/android-slave/Dockerfile

```diff
@@ -16,13 +16,62 @@ RUN yum install -y centos-release-scl-rh && \
     INSTALL_PKGS="rh-nodejs4 rh-nodejs4-npm rh-nodejs4-nodejs-nodemon" && \
     ln -s /usr/lib/node_modules/nodemon/bin/nodemon.js /usr/bin/nodemon && \
     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS
+
+# Install Java
+RUN INSTALL_PKGS="java-1.8.0-openjdk-devel" && \
+    yum install -y --enablerepo=centosplus $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all -y
 
 RUN chown -R 1001:0 $HOME && \
     chmod -R g+rw $HOME
 
-## TODO Extend this to include android build tools
-## See https://github.com/appunite/docker/tree/master/android-java8-r25 
+# Install build dependencies
+RUN yum install -y git wget python curl 
+## Support native builds
+# RUN yum install gcc gcc-c++ make openssl-devel
+
+# Copy install tools
+COPY tools /opt/tools
+ENV PATH ${PATH}:/opt/tools
+
+# Install Android SDK
+RUN cd /opt && wget --output-document=android-sdk.tgz --quiet https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz && \
+  tar xzf android-sdk.tgz && \
+  rm -f android-sdk.tgz
+
+RUN yum install -y expect expectk
+
+RUN cd /opt && chown -R root.root android-sdk-linux && \
+  chmod -R 777 /opt/tools && \
+  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools" && \
+  /opt/tools/android-accept-licenses.sh "android-sdk-linux/tools/android update sdk --all --no-ui --filter platform-tools,tools,build-tools-21.0.0,build-tools-21.0.1,build-tools-21.0.2,build-tools-21.1.0,build-tools-21.1.1,build-tools-21.1.2,build-tools-22.0.0,build-tools-22.0.1,build-tools-23.0.0,build-tools-23.0.3,build-tools-24.0.0,build-tools-24.0.1,build-tools-24.0.2,build-tools-24.0.3,build-tools-25.0.0,android-21,android-22,android-23,android-24,android-25,addon-google_apis_x86-google-21,extra-android-support,extra-android-m2repository,extra-google-m2repository,extra-google-google_play_services,sys-img-armeabi-v7a-android-24"
+
+# Setup environment
+ENV ANDROID_HOME /opt/android-sdk-linux
+ENV PATH ${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools
+
+RUN which adb
+RUN which android
+
+RUN android list targets
+
+# Create emulator
+# RUN echo "no" | android create avd \
+#                 --force \
+#                 --device "Nexus 5" \
+#                 --name test \
+#                 --target android-24 \
+#                 --abi armeabi-v7a \
+#                 --skin WVGA800 \
+#                 --sdcard 512M
+
+# Install Cordova
+#RUN npm install -g cordova
```

Comment:
Running and installing cordova directly there has many drawbacks and I do not feel we should do it this way. 
Locking up image to one version may be really bad idea. Suggestions:

- npm install -g cordova in Jenkins file (longer builds)
- Separate docker image that is basing on this image with cordova installed (more hassle to update)
- Docker volume with multiple cordova command line tools installed and some support in jenkins file to "mount" proper version on demand (probably to large to be done initially)